### ISSUE TITLE
logind: Lingering is incompatible with homed users

### ISSUE THREAD

**Issue by AdrianVovk** [systemd/systemd#34151]
Split out from https://github.com/systemd/systemd/issues/33541, see there for background and context.

> First issue exposed here: logind lingering outright _cannot_ work with systemd-homed users. At least currently. Maybe eventually we'll have TPM based authentication methods for homed users, and will be able to use the TPM to unlock a home dir for lingering. But even then, that configuration will need to come from homed, not from logind. `loginctl enable-linger <some-homed-user>` can never work. Maybe we should just outright refuse to turn on linger for homed users via the command line, and ignore any homed users that might have previously been configured in /var/lib/systemd/linger.
>
> logind should also log that it's activating a user w/ linger... I suspect there won't ever be enough lingering users on a system fro a long message about it to be too noisy. We spent a long time trying to figure out what, exactly, is trying to activate the `user@<uid>.service` at boot!

**Comment by poettering** [comment]
I was thinking of teach the PAM logic in the service manager to ask passwords via the `systemd-ask-password` API. that would mean if you turn on lingering for a homed user, plymouth would ask for the relevant info at boot.

**Comment by YHNdnzj** [comment]
I agree with @AdrianVovk here. TBH I think lingering itself is a pretty terrible/unmanageable concept, and especially for homed users such concept should really be thrown away. It requires user credentials to be provided at boot, and it would hold the home directory open even after all active sessions exit. That kind of contradicts the goal of homed completely.

**Comment by poettering** [comment]
Maybe we should add a new field to user records: `accessRequiresAuthentication` in the `status` section or so. Would be a boolean, defaults to false. If set to true (which homed would set implicitly), we'd refuse enabling lingering in logind.

**Comment by poettering** [comment]
btw, one more clarification: lingering is two things actually. one one hand it's that user@.service is started immediately at boot for a user. and on the other hand it's that user@.service is left running after the user logged out last. I wonder if we should safely support the latter even on homed, in case lingering is enabled. There's no real reason to prohibit that.

**Comment by YHNdnzj** [comment]
> on the other hand it's that user@.service is left running after the user logged out last.

Like I mentioned above, I think this also mostly defeats the purpose of homed, which shall evict the encryption key when the user logs out...

**Comment by crepererum** [comment]
> > on the other hand it's that [user@.service](mailto:user@.service) is left running after the user logged out last.
> 
> Like I mentioned above, I think this also mostly defeats the purpose of homed, which shall evict the encryption key when the user logs out...

I think leaving the session intact after logout has a valid use case: servers/IoT devices. The system can boot up without external interaction and establish a network connection and then a user can login to provide encryption keys for their home directory. Ideally you would have a TPM or something to do that, but many devices (especially single board computers) don't provide that. This way, you at least have encryption at rest, so if the device is stolen or seized, the user data isn't leaked.
